### ISSUE TITLE
Unexpected differences between GLMakie and CairoMakie

### ISSUE THREAD

**Issue by briochemc** [JuliaPlots/Makie.jl#666]
Here is a MWE:
```julia
using AbstractPlotting
using AbstractPlotting.MakieLayout
# Some data
x, y = 0:0.01:2π, 0:0.1:10
f(x,y) = (cos(x) + y > 9) ? NaN : cos(x) * sin(y)
data = [f(x,y) for x in x, y in y] ;
# A function to plot
function myplot(x, y, data)
    scene, layout = layoutscene(10, resolution = (300, 300))
    ax = layout[1,1] = LAxis(scene)
    heatmap!(ax, x, y, data)
    ylims!(ax, (10,0))
    return scene
end

# GLMakie plot
using GLMakie; GLMakie.activate!()
display(myplot(x, y, data))

# CairoMakie plot
using CairoMakie; CairoMakie.activate!()
save("myplot.png", myplot(x, y, data))
```

which shows plots this with GLMakie (left) and saves this with CairoMakie (right):
<p>
  <img width="262" src="https://user-images.githubusercontent.com/4486578/89009670-5bd8ca00-d350-11ea-97ff-5ed91955afd2.png">
  <img width="262" src="https://user-images.githubusercontent.com/4486578/89009125-457e3e80-d34f-11ea-98cd-a761ede66fc7.png">
</p>

I see a couple of issues with how the figure is saved with CairoMakie here:
- The *y*-axis is not flipped despite the `ylims!(ax, (10,0))` line, so the wavy boundary of `NaN`s is at the top instead of at the bottom.
- The wavy boundary of `NaN`s is filled with the "lowest" color of the colormap instead of being white/transparent.

How can I solve these two things so that the CairoMakie-saved file shows the same as the GLMakie plot?

**Comment by briochemc** [comment]
FWIW, I also tried the `nan_color` attribute (https://github.com/JuliaPlots/Makie.jl/issues/527), which works with GLMakie's display but not with CairoMakie's saved figure for me.

**Comment by jkrumbiegel** [comment]
I'll have a look, seems to be that the heatmap implementation bypasses the axis projection (probably only based on the rectangle it's in), thereby not flipping correctly. nan_color is probably just not implemented at all, rather than being buggy

**Comment by briochemc** [comment]
OK, so thanks to @jkrumbiegel who fixed it in [CairoMakie v0.3.1](https://github.com/JuliaPlots/CairoMakie.jl/releases/tag/v0.3.1), after some pkg updating, I get exactly the same with both! (GLMakie on the left, CairoMakie on the right)



<p>
  <img  width=30% src="https://user-images.githubusercontent.com/4486578/89366637-98267480-d71a-11ea-8973-e8e9391e7c7c.png">
  <img  width=30% src="https://user-images.githubusercontent.com/4486578/89366581-7c22d300-d71a-11ea-808f-0c088dec5bf6.png">
</p>
